tests/test_TestForMissingProperties.py

```diff
@@ -116,18 +116,30 @@ def find_feature_in_code(self, all_python_files, feature):
 
     def read_all_python_code(self):
         python_path = join(dirname(__file__), '../PyViCare')
-        python_files = [f for f in listdir(python_path) if isfile(join(python_path, f))]
+        # stores pairs of [path, filename]
+        python_files = []
+        # searches in all subdirectories
+        self.get_all_files(python_files, python_path)
 
         all_python_files = {}
 
         for python in python_files:
-            if not python.endswith(".py"):
+            if not python[1].endswith(".py"):
                 continue
 
-            with open(join(python_path, python)) as f:
-                all_python_files[python] = f.read()
+            with open(join(python[0], python[1])) as f:
+                all_python_files[python[1]] = f.read()
+
         return all_python_files
 
+    def get_all_files(self, files, path):
```

Comment:
Please return a new array, instead of passing them as an argument. Or at least provide a second method which hides this and rename this method to include _recursive the more more self explanatory.